### PR TITLE
filter listing on the basis of category

### PR DESCRIPTION
**Pull request by vikas-cis**
#881

### CODE REVIEW

File: src/app/market/overview-listings/overview-listings.component.ts [hunk 1]
```diff
@@ -170,6 +172,10 @@ export class OverviewListingsComponent implements OnInit, OnDestroy {
     return this.pages[0].pageNumber;
   }
 
+  filter(): void {
+    this.loadPage(1)
```
Comment (kewde) [inline src/app/market/overview-listings/overview-listings.component.ts]: clearAndLoadPage()?

File: src/app/market/overview-listings/overview-listings.component.ts [hunk 2]
```diff
@@ -85,16 +86,17 @@ export class OverviewListingsComponent implements OnInit, OnDestroy {
...
   }
 
   loadPage(pageNumber: number, clear?: boolean) {
     const max = this.pagination.maxPerPage;
 
     const search = this.filters.search;
-
-    this.listingService.search(pageNumber, max, null, search).take(1).subscribe((listings: Array<any>) => {
+    console.log(this.filters.sort)
+    this.listingService.search(pageNumber, max, null, search, this.filters.sort).take(1).subscribe((listings: Array<any>) => {
```
Comment (kewde) [inline src/app/market/overview-listings/overview-listings.component.ts]: move it to
const sort = this.filters.sort to shorten things up

**Comment by kewde** [comment]
also completes #863

File: src/app/market/overview-listings/overview-listings.component.html [hunk 3]
```diff
@@ -15,9 +15,9 @@
         <mat-option *ngFor="let country of countryList?.countries" [value]="country.name">{{ country.name }}</mat-option>
       </mat-select>
 
-      <mat-select placeholder="Categories" [formControl]="categories" class="filter category TO_DO">
+      <mat-select placeholder="All categories" (change)="clearAndLoadPage()" [(ngModel)]="filters.category" [formControl]="categories" class="filter category">
```
Comment (kewde) [inline src/app/market/overview-listings/overview-listings.component.html]: Both formcontrol and ngmodel, make a pick, don't use both.

**Comment by rynomster** [comment]
Has conflcts

**Pull request merged**